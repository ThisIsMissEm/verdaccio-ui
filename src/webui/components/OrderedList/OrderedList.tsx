/**
 * @prettier
 */

import React, { ReactNode } from 'react';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/styles';

interface Props {
  children: ReactNode;
}

const useStyles = makeStyles({
  listItem: {
    alignItems: 'flex-start',
  },
  interpunct: {
    paddingRight: 8,
  },
});

const OrderedList: React.FC<Props> = ({ children }) => {
  const classes = useStyles();
  return (
    <List>
      {React.Children.map(children, (child, index) => (
        <ListItem key={index} className={classes.listItem}>
          <Typography variant="body2" className={classes.interpunct}>{`${index + 1}.`}</Typography>
          {child}
        </ListItem>
      ))}
    </List>
  );
};
export default OrderedList;
