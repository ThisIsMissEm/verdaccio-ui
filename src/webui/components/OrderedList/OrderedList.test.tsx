/**
 * @prettier
 */

import React from 'react';
import { shallow } from 'enzyme';
import OrderedList from '.';
import Typography from '@material-ui/core/Typography';

const listExample = ['first', 'second', 'third'];

describe('<OrderedList /> component', () => {
  test('should render the component in default state', () => {
    const wrapper = shallow(
      <OrderedList>
        {listExample.map(itemExample => (
          <Typography key={itemExample}>{itemExample}</Typography>
        ))}
      </OrderedList>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
