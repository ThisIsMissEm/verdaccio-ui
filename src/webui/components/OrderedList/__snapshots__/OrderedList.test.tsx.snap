// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderedList /> component should render the component in default state 1`] = `"<ul class=\\"MuiList-root-1 MuiList-padding-2\\"><li class=\\"MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 Hook-listItem-pb3m1\\"><p class=\\"MuiTypography-root-17 MuiTypography-body2-25 Hook-interpunct-1jg88t5\\">1.</p><p class=\\"MuiTypography-root-17 MuiTypography-body1-26\\">first</p></li><li class=\\"MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 Hook-listItem-pb3m1\\"><p class=\\"MuiTypography-root-17 MuiTypography-body2-25 Hook-interpunct-1jg88t5\\">2.</p><p class=\\"MuiTypography-root-17 MuiTypography-body1-26\\">second</p></li><li class=\\"MuiListItem-root-5 MuiListItem-default-8 MuiListItem-gutters-13 Hook-listItem-pb3m1\\"><p class=\\"MuiTypography-root-17 MuiTypography-body2-25 Hook-interpunct-1jg88t5\\">3.</p><p class=\\"MuiTypography-root-17 MuiTypography-body1-26\\">third</p></li></ul>"`;
